frontend/app/views/spree/products/_thumbnails.html.erb

```diff
@@ -1,17 +1,19 @@
 <%# no need for thumbnails unless there is more than one image %>
-<% if (@product.images + @product.variant_images).uniq.size > 1 %>
+<% if @product.gallery.images.size > 1 %>
   <ul id="product-thumbnails" class="thumbnails inline" data-hook>
-    <% @product.images.each do |i| %>
-      <li class='tmb-all tmb-<%= i.viewable.id %>'>
-        <%= link_to(image_tag(i.attachment.url(:mini)), i.attachment.url(:product)) %>
+
+    <% @product.gallery.images.each do |image| %>
+      <% next if image.viewable_id != @product.master.id %>
```

Comment:
I don't see how I am going to be able to write this `_thumbnails` partial using only image URLs. At some point an image is going to need the context of which variant it is for. What do you think, @tvdeyen?

If we have situations like this (and the `admin/_image_row` partial), I think it would make sense for the galleries to return an "Image" class. If a developer wants to drop in a different implementation they can provide their own image class that responds to `#url`, `#id`, `#viewable_id`, and `#alt`. I was thinking something like this:

```
class MyImageClass
  def initialize(image)
    @image = image
  end

  def url(size)
    @image.the_way_this_image_library_gets_urls(size)
  end

  # etc...
end

class MyVariantGallery
  def images
    however_the_images_are_stored.map { |image| MyImage.new(image) }
  end
end
```

Reply:
I like the idea of a separate `Image` class 👌

So we would have a repository (`Gallery`) and a presenter (maybe `Picture`?). This sounds like a good plan to me.

Reply:
After thinking about this, I think that we can transition `Spree::Image` to function as this presenter. I don't think there is much value in adding another class (`Spree::Picture`), when we can use `Spree::Image` to do the same thing. I've updated the PR to reflect this.